### PR TITLE
[hotfix][docs] Fix Apache Avro Specification Link.

### PR DESCRIPTION
**Pull request by donPain**
Just correcting the url reference to access the Avro specification.

### CODE REVIEW

**Comment by davidradl** [comment]
Please add [hotfix] at the start of the title as per the process when there is no associated Jira.
Please could you squash the commits.
Please could you do the Chinese version as well.

**Comment by donPain** [comment]
@davidradl I made the changes, could you review it again?

File: docs/content/docs/connectors/table/formats/avro.md [hunk 1]
```diff
@@ -218,4 +218,4 @@ So the following table lists the type mapping from Flink type to Avro type.
 
 In addition to the types listed above, Flink supports reading/writing nullable types. Flink maps nullable types to Avro `union(something, null)`, where `something` is the Avro type converted from Flink type.
 
-You can refer to [Avro Specification](https://avro.apache.org/docs/current/spec.html) for more information about Avro types.
+You can refer to [Avro Specification](https://avro.apache.org/docs/++version++/specification/) for more information about Avro types.
```
Comment (davidradl) [inline docs/content/docs/connectors/table/formats/avro.md:221]: looks good - I notice that there is. reference in  parquet.md as well to the spec for 1.10. should we align that link here as well?

Comment (donPain) [reply docs/content/docs/connectors/table/formats/avro.md:221]: Good point, I changed all values ​​from version "1.10.0" to "++version++".

**Comment by snuyanzin** [comment]
can you also change it at https://github.com/apache/flink/blob/72b32fcd53f2d0d81329298a6c0a7d08e084f769/flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/typeutils/AvroSerializerSnapshot.java#L169

**Comment by donPain** [comment]
> can you also change it at
> 
> https://github.com/apache/flink/blob/72b32fcd53f2d0d81329298a6c0a7d08e084f769/flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/typeutils/AvroSerializerSnapshot.java#L169
Sure, I made the changes.

**Comment by snuyanzin** [comment]
run `./mvnw spotless:apply`
before
```
git add . && git commit
```
as mentioned at https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=64325&view=logs&j=52b61abe-a3cc-5bde-cc35-1bbe89bb7df5&t=54421a62-0c80-5aad-3319-094ff69180bb&l=12927  (the reason of failure)

**Review by davidradl — APPROVED** [review]

**Comment by davidradl** [comment]
Reviewed by Chi on 12/12/24 Approve - looking for committer to merge

File: docs/content.zh/docs/connectors/datastream/formats/parquet.md [hunk 2]
```diff
@@ -198,14 +198,14 @@ ds = env.from_source(source, WatermarkStrategy.no_watermarks(), "file-source")
...
+- [Generic record](https://avro.apache.org/docs/++version++/api/java/index.html)
+- [Specific record](https://avro.apache.org/docs/++version++/api/java/index.html)
+- [Reflect record](https://avro.apache.org/docs/++version++/api/java/org/apache/avro/reflect/package-summary.html)
 
 ### Generic record
 
-使用 JSON 定义 Avro schemas。你可以从 [Avro specification](https://avro.apache.org/docs/1.10.0/spec.html) 获取更多关于 Avro schemas 和类型的信息。
-此示例使用了一个在 [official Avro tutorial](https://avro.apache.org/docs/1.10.0/gettingstartedjava.html) 中描述的示例相似的 Avro schema：
+使用 JSON 定义 Avro schemas。你可以从 [Avro specification](https://avro.apache.org/docs/++version++/spec.html) 获取更多关于 Avro schemas 和类型的信息。
+此示例使用了一个在 [official Avro tutorial](https://avro.apache.org/docs/++version++/gettingstartedjava.html) 中描述的示例相似的 Avro schema：
```
Comment (rmetzger) [inline docs/content.zh/docs/connectors/datastream/formats/parquet.md:208]: This link seems broken

**Comment by rmetzger** [comment]
Flink is using Avro 1.11.4. Wouldn't it make more sense to link to the docs of that version, then add a note to the pom file where the avro version is defined to make sure folks update the docs when updating the avro version in the pom file?
Ref: https://github.com/apache/flink/blob/release-1.20/pom.xml#L149

Also, I believe our docs system supports variables -- it would be much easier to have one variable defining the avro version, instead of having this spread all over the place.

**Comment by donPain** [comment]
> Flink is using Avro 1.11.4. Wouldn't it make more sense to link to the docs of that version, then add a note to the pom file where the avro version is defined to make sure folks update the docs when updating the avro version in the pom file? Ref: https://github.com/apache/flink/blob/release-1.20/pom.xml#L149
> 
> Also, I believe our docs system supports variables -- it would be much easier to have one variable defining the avro version, instead of having this spread all over the place.

Apparently most of the avro documentation has broken links above 1.13.0.

Could you give me a reference on how to use variables to set the url in the documentation?

**Comment by donPain** [comment]
@flinkbot run azure

**Comment by davidradl** [comment]
@rmetzger @donPain I had a quick look at the docs, it seems that Hugo does allow templating which supports variables. I can see the Flink code referencing the predefined variables with {{  }}. I see `{{< all_versions >}}  ` which looks like a custom variable; but I am not sure how it is populated. I see code bringing in the different connectors. I have not yet found how we would want to add global variables like this.

**Comment by davidradl** [comment]
@rmetzger as discussed at the Chi meeting are you ok to merge this? @donPain can you raise a separate issue to track the variable replacement improvement idea please.

**Review by rmetzger — APPROVED** [review]
Thanks for working on this.
Let's intro the variable in a separate JIRA / PR.

**Comment by rmetzger** [comment]
@donPain could you raise a PR forward-porting the commit to the `master` branch as well? Thx

**Comment by donPain** [comment]
> rmetzger

Sure, here it is: https://github.com/apache/flink/pull/25850

**Pull request merged**